Sources/EdgeNetworkHandlers/ResponseCallbackHandler.swift

```diff
@@ -39,32 +51,93 @@ class ResponseCallbackHandler {
         responseHandlers[requestEventId] = unwrappedResponseHandler
     }
 
-    /// Unregisters a `EdgeResponseHandler` for the specified `requestEventId`. After this operation,
-    /// the associated response handler is not invoked anymore.
+    /// Registers a completion handler for the specified `requestEventId`. This handler is invoked when the entire Edge response content has been
+    /// handled entirely by the Edge extension, containing a list of `EdgeEventHandle` and a list of `EdgeEventError`. These lists can be empty or
+    /// can contain one or multiple items based on the request and the server side response.
+    ///
+    /// - Parameters:
+    ///   - requestEventId: unique event identifier for which the response callback is registered; should not be empty
+    ///   - completionHandler: the completion handler that needs to be registered, should not be nil
+    func registerCompletionHandler(requestEventId: String, completionHandler: (([EdgeEventHandle], [EdgeEventError]) -> Void)?) {
+        guard let unwrappedCompletion = completionHandler else { return }
+        guard !requestEventId.isEmpty else {
+            Log.warning(label: TAG, "Failed to register completion handler because of empty request event id.")
+            return
+        }
+
+        completionHandlers[requestEventId] = unwrappedCompletion
+    }
+
+    /// Calls onComplete and unregisters a `EdgeResponseHandler` for the specified `requestEventId`. After this operation,
+    /// the associated response handler or completion handler is removed and no longer called.
     /// - Parameter requestEventId: unique event identifier for experience events; should not be empty
-    func unregisterResponseHandler(requestEventId: String) {
+    func unregisterCallbacks(requestEventId: String) {
         guard !requestEventId.isEmpty else { return }
-
-        if responseHandlers[requestEventId] != nil {
+        if let responseHandler = responseHandlers[requestEventId] {
+            responseHandler.onComplete()
             responseHandlers[requestEventId] = nil
-            Log.trace(label: TAG, "Removing callback for Edge response with request unique id \(requestEventId).")
         }
+
+        if let completionHandler = completionHandlers[requestEventId] {
+            completionHandler(edgeEventHandles[requestEventId] ?? [], edgeEventErrors[requestEventId] ?? [])
+            completionHandlers[requestEventId] = nil
+        }
+
+        edgeEventHandles[requestEventId] = nil
+        edgeEventErrors[requestEventId] = nil
+
+        Log.trace(label: TAG, "Removing completion handlers for Edge response with request unique id \(requestEventId).")
```

Comment:
nit: As this is for either completion handlers or response handlers, maybe reword to "Removing callback handlers for Edge response ..." ?